### DISCUSSION TITLE
Does anyone know what was accomplished in comment https://github.com/dotnet/core/issues/7556#issuecomment-1786377918

### DISCUSSION THREAD

**Discussion by Sin-Shadow-Fox** [dotnet/runtime#104200] [Q&A]
Does anyone know what they're talking about or what these C++ runtimes are?

https://github.com/dotnet/core/issues/7556#issuecomment-1786377918

**Comment by dotnet-policy-service** [comment]
Tagging subscribers to this area: @dotnet/area-meta
See info in [area-owners.md](https://github.com/dotnet/runtime/blob/main/docs/area-owners.md) if you want to be subscribed.

**Comment by danmoseley** [comment]
@agocke

**Comment by KalleOlaviNiemitalo** [comment]
@Sin-Shadow-Fox, there's some explanation at [Microsoft Visual C++ Redistributable latest supported downloads](<https://learn.microsoft.com/cpp/windows/latest-supported-vc-redist> "Latest supported Visual C++ Redistributable downloads | Microsoft Learn"). The page says "Many applications built using Microsoft C and C++ tools require these libraries"; the .NET Runtime is built for Windows using those tools.

**Comment by Sin-Shadow-Fox** [comment]
> @Sin-Shadow-Fox, there's some explanation at [Microsoft Visual C++ Redistributable latest supported downloads](https://learn.microsoft.com/cpp/windows/latest-supported-vc-redist). The page says "Many applications built using Microsoft C and C++ tools require these libraries"; the .NET Runtime is built for Windows using those tools.

While I get what your saying, I don't understand how that explains the comment I linked to. It sounds like someone managed to get .net 8 to run on windows 7 and I'd really like to figure out how.

**Comment by KalleOlaviNiemitalo** [comment]
The comment says

> I can use .NET 7 event maybe .NET 8 on windows 7

and the screen shot shows only .NET 7 on Windows 7, so I'm not sure they managed to run .NET 8 on Windows 7.

**Comment by Sin-Shadow-Fox** [comment]
> The comment says
> 
> > I can use .NET 7 event maybe .NET 8 on windows 7
> 
> and the screen shot shows only .NET 7 on Windows 7, so I'm not sure they managed to run .NET 8 on Windows 7.

Neither am I but I can't exactly ask either of them how things went as dotnet locked that whole thread and limited conversation to collaborators and I don't have the permission to re-open it (or know how to become a collaborator).

**Reply by xparadoxical** [reply]
@RockNHawk 🤷‍♂️

**Comment by jkotas** [comment]
.NET 8 is **unsupported** on Windows 7.

.NET 8 is not actively blocked from running on Windows 7. Some .NET 8 apps will run on Windows 7, some .NET 8 apps are not going to run. YMMV.

**Comment by Sin-Shadow-Fox** [comment]
> .NET 8 is **unsupported** on Windows 7.
> 
> .NET 8 is not actively blocked from running on Windows 7. Some .NET 8 apps will run on Windows 7, some .NET 8 apps are not going to run. YMMV.

Well I'm not asking if .NET 8 is supported on windows 7.
I'm asking for help understanding what the people in the comment i linked to are talking about because it sounds like they got .NET 8 working on windows 7.

**Comment by Sin-Shadow-Fox** [comment]
P.S. request to un-close my issue as my question wasn't answered yet.

**Comment by jkotas** [comment]
> I'm asking for help understanding what the people in the comment i linked to are talking about because it sounds like they got .NET 8 working on windows 7.

Yes, they got it working for their app by installing the vc_redist. The comment is very clear about that.

I tried to explain why it may not be sufficient for the app that you are trying to run.

> P.S. request to un-close my issue as my question wasn't answered yet.

I am going to convert this to a discussion.

**Reply by Sin-Shadow-Fox** [reply]
> > I'm asking for help understanding what the people in the comment i linked to are talking about because it sounds like they got .NET 8 working on windows 7.
> 
> Yes, they got it working for their app by installing the vc_redist. The comment is very clear about that.

Well yes but it's not clear on why it worked or if it worked after other steps.
I'd like to not go installing random things on my computer and hoping it fixes the problem, especially with the number of people here saying randomly installing things can be dangerous. I hope you understand why i'm being so cuatous and seeking step by step guidance.

> > P.S. request to un-close my issue as my question wasn't answered yet.
> 
> I am going to convert this to a discussion.

Thank you.

**Reply by tannergooding** [reply]
The general consideration is that there are no guarantees here. What works for one app may not work for another.

Officially Windows 7 is an unsupported environment (having gone out of mainstream support 9.5 years ago and out of extended support 1.5 years ago), so exactly what is required to get any given app working is going to be application specific. It may be as simple as installing an appropriate VC++ runtime redistributable, it may not ever work due to executing some code path that assumes a supported OS (such as Windows 10) is being run against and using an API that only exists on the such supported OS.

Your mileage may vary and there isn't going to be one quick fix or general purpose solution. The proper thing to do is to move forward to a supported OS version, which will ensure you don't hit such issues in the first place. -- This is particularly prevalent given that Windows 7 is no longer getting updates and thus you may be more vulnerable to security risks and viruses (https://support.microsoft.com/en-us/windows/windows-7-support-ended-on-january-14-2020-b75d4580-2cc7-895a-2c9c-1466d9a53962)

**Reply by Sin-Shadow-Fox** [reply]
> The general consideration is that there are no guarantees here. What works for one app may not work for another.
> 
> Exactly what is required to get any given app working is going to be application specific. It may be as simple as installing an appropriate VC++ runtime redistributable, it may not ever work due to executing some code path that assumes a supported OS (such as Windows 10) is being run against and using an API that only exists on the such supported OS.

Hence the need for assistance.

> Your mileage may vary and there isn't going to be one quick fix or general purpose solution. The proper thing to do is to move forward to a supported OS version, which will ensure you don't hit such issues in the first place. 

As the only real option for updating OSs is Linux, I'm not fond of this idea as Linux is a very teck savvy OS that requires a lot of technical expertise and knowledge. Besides, usually when i hit a road block, the problem is fixable and doesn't require something as drastic as changing OSs.

> This is particularly prevalent given that Windows 7 is no longer getting updates and thus you may be more vulnerable to security risks and viruses (https://support.microsoft.com/en-us/windows/windows-7-support-ended-on-january-14-2020-b75d4580-2cc7-895a-2c9c-1466d9a53962)

Actually this is just propaganda by microsoft to force people off windows 7 because windows 7 does not incorporate microsoft's close system (kinda why a lot of us are staying on it, well that and it works a lot better than 10/11). Windows 7 is no more vulnerable than 10/11, this has already been proven. Please do not believe everything you read online, especially if it's from micrososft.

**Reply by agocke** [reply]
> Hence the need for assistance.

I want to clarify that “unsupported” means Microsoft will not provide assistance in this matter. You may find a community member to help you, however.

**Reply by Sin-Shadow-Fox** [reply]
> > Hence the need for assistance.
> 
> I want to clarify that “unsupported” means Microsoft will not provide assistance in this matter. You may find a community member to help you, however.

That's my goal ^_^

**Comment by Sin-Shadow-Fox** [comment]
Whoever marked jkotas's [comment](https://github.com/dotnet/runtime/discussions/104200#discussioncomment-9913318) as the answer, please keep in mind that my question is NOT "is .NET 8 supported in windows 7". Please reread the post and subsequent comments carefully.

**Reply by teo-tsirpanis** [reply]
@Sin-Shadow-Fox to more closely answer your question (which was already partially done in https://github.com/dotnet/runtime/discussions/104200#discussioncomment-9913315), .NET has a dependency to the Microsoft Visual C++ redistributables, which are preinstalled in Windows 10+, but for earlier Windows versions, you would have to install these dependencies by yourself.

I hope this explanation is sufficient. Let me know if you have any further questions.

**Reply by Sin-Shadow-Fox** [reply]
@teo-tsirpanis It does explain the need for downloading VC++ redist and why this would solve [RockNHawk](https://github.com/RockNHawk)'s specific problem. Unfortunately this makes it clear that their solution will probably not help me. Would you be willing to help me try to get .NET8/SteamRE to run on my computer?

**Reply by teo-tsirpanis** [reply]
I presume you have installed the Visual C++ redistributables, and have downloaded the executable from https://github.com/SteamRE/DepotDownloader/releases/tag/DepotDownloader_2.6.0, right? If yes, what is the error you are encountering?

Also keep in mind that if your computer runs Windows 7, it is likely that you will sooner or later encounter a non-trivial issue that will not be fixed because Windows 7 is not supported on .NET 8.

**Reply by Sin-Shadow-Fox** [reply]
> I presume you have installed the Visual C++ redistributables, and have downloaded the executable from https://github.com/SteamRE/DepotDownloader/releases/tag/DepotDownloader_2.6.0, right? If yes, what is the error you are encountering?

I . . . have not. Again, i didn't want to install anything i haven't explicitly been told to (especially if it isn't meant for me). I actually have another [discussion](https://github.com/dotnet/runtime/discussions/103779) in this repository asking which variant of a .NET version i should install which still hasn't been answered. 

> Also keep in mind that if your computer runs Windows 7, it is likely that you will sooner or later encounter a non-trivial issue that will not be fixed because Windows 7 is not supported on .NET 8.

I doubt it. Windows 7 is built very well (much to microsoft's chagrin) and i personally believe that windows 7 is going to make a resurgence in the tech community if not outright diverge into a new OS altogether. There are already a lot of people working on providing windows 7 updates to keep it working with "noncompatible" programs. There is even a kernel extender for it now.

**Reply by teo-tsirpanis** [reply]
I am explicitly telling you to install [these dependencies for Windows 7](https://learn.microsoft.com/en-us/dotnet/core/install/windows#windows-7--81--server-2012). 😛 Don't worry, they have been installed in millions of computers, and the installers will be digitally signed.

> which variant of a .NET version i should install which still hasn't been answered

Actually, if you install https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.6.0/DepotDownloader-windows-x64.zip, you don't need any .NET runtime installed because it is a self-contained executable (notice that it is significantly bigger than the framework-dependent `DepotDownloader-framework.zip`).

**Reply by Sin-Shadow-Fox** [reply]
> I am explicitly telling you to install [these dependencies for Windows 7](https://learn.microsoft.com/en-us/dotnet/core/install/windows#windows-7--81--server-2012). 😛 Don't worry, they have been installed in millions of computers, and the installers will be digitally signed.

I tried to but i got this message
![Capture](https://github.com/dotnet/runtime/assets/96899282/37ac10eb-9daf-4d34-88b3-31c4e6e8fc58)


> > which variant of a .NET version i should install which still hasn't been answered
> 
> Actually, if you install https://github.com/SteamRE/DepotDownloader/releases/download/DepotDownloader_2.6.0/DepotDownloader-windows-x64.zip, you don't need any .NET runtime installed because it is a self-contained executable (notice that it is significantly bigger than the framework-dependent `DepotDownloader-framework.zip`).

When i open DepotDownloader.exe, it just opens command prompt for a second and then closes.

**Reply by colejohnson66** [reply]
The error message says what the problem is. You already have a newer version already installed, so you can skip that dependency.

**Reply by Sin-Shadow-Fox** [reply]
> The error message says what the problem is. You already have a newer version already installed, so you can skip that dependency.

Sweet. I thought that message was an error. You wouldn't believe how many times I've received an error message that itself was also an error and gave the incorrect reason for why something failed.

Now i just need to figure out what's happening with the DepotDownloader.exe.

**Reply by teo-tsirpanis** [reply]
> When i open DepotDownloader.exe, it just opens command prompt for a second and then closes.

I think that's what happened to me when I tried opening it. DepotDownloader is a command-line app. Can you try opening a command line and running `DepotDownloader.exe` from there? Does it display an error message?

**Reply by Sin-Shadow-Fox** [reply]
> Can you try opening a command line and running `DepotDownloader.exe` from there?

How do i do that?

**Reply by Sin-Shadow-Fox** [reply]
@teo-tsirpanis Hello?

**Reply by teo-tsirpanis** [reply]
You can try the following steps:

1. Press <kbd>Winkey</kbd>+<kbd>R</kbd> on your keyboard.
2. Type `cmd` and press OK.
3. Type on the command line the path to `DepotDownloader.exe` and press <kbd>Enter</kbd>.

**Reply by Sin-Shadow-Fox** [reply]
> You can try the following steps:
> 
>     1. Press Winkey+R on your keyboard.
> 
>     2. Type `cmd` and press OK.
> 
>     3. Type on the command line the path to `DepotDownloader.exe` and press Enter.

Microsoft Windows [Version 6.1.7601]
Copyright (c) 2009 Microsoft Corporation.  All rights reserved.

C:\Users\User>C:\Users\User\Documents\Downloads\DepotDownloader-windows-x64\Depo
tDownloader.exe

Usage - downloading one or all depots for an app:
        depotdownloader -app <id> [-depot <id> [-manifest <id>]]
                [-username <username> [-password <password>]] [other options]

Usage - downloading a workshop item using pubfile id
        depotdownloader -app <id> -pubfile <id> [-username <username> [-password
 <password>]]
Usage - downloading a workshop item using ugc id
        depotdownloader -app <id> -ugc <id> [-username <username> [-password <pa
ssword>]]

Parameters:
        -app <#>                                - the AppID to download.
        -depot <#>                              - the DepotID to download.
        -manifest <id>                  - manifest id of content to download (re
quires -depot, default: current for branch).
        -beta <branchname>                      - download from specified branch
 if available (default: public).
        -betapassword <pass>            - branch password if applicable.
        -all-platforms                  - downloads all platform-specific depots
 when -app is used.
        -os <os>                                - the operating system for which
 to download the game (windows, macos or linux, default: OS the program is curre
ntly running on)
        -osarch <arch>                          - the architecture for which to
download the game (32 or 64, default: the host's architecture)
        -all-languages                          - download all language-specific
 depots when -app is used.
        -language <lang>                                - the language for which
 to download the game (default: english)
        -lowviolence                            - download low violence depots w
hen -app is used.

        -ugc <#>                                - the UGC ID to download.
        -pubfile <#>                    - the PublishedFileId to download. (Will
 automatically resolve to UGC id)

        -username <user>                - the username of the account to login t
o for restricted content.
        -password <pass>                - the password of the account to login t
o for restricted content.
        -remember-password              - if set, remember the password for subs
equent logins of this user. (Use -username <username> -remember-password as logi
n credentials)

        -dir <installdir>               - the directory in which to place downlo
aded files.
        -filelist <file.txt>    - a list of files to download (from the manifest
). Prefix file path with 'regex:' if you want to match with regex.
        -validate                               - Include checksum verification
of files already downloaded

        -manifest-only                  - downloads a human readable manifest fo
r any depots that would be downloaded.
        -cellid <#>                             - the overridden CellID of the c
ontent server to download from.
        -max-servers <#>                - maximum number of content servers to u
se. (default: 20).
        -max-downloads <#>              - maximum number of chunks to download c
oncurrently. (default: 8).
        -loginid <#>            - a unique 32-bit integer Steam LogonID in decim
al, required if running multiple instances of DepotDownloader concurrently.

C:\Users\User>

**Reply by Sin-Shadow-Fox** [reply]
@teo-tsirpanis

**Reply by teo-tsirpanis** [reply]
The program has printed its help note, which means it did not catastrophically fail on Windows 7.

As stated by previous comments to this discussion, a .NET 8 app might fail on Windows 7 or might not but if it does fail, Microsoft will not provide any form of support.

Let me know if you have another question.

**Reply by Sin-Shadow-Fox** [reply]
> The program has printed its help note, which means it did not catastrophically fail on Windows 7.
> 
> As stated by previous comments to this discussion, a .NET 8 app might fail on Windows 7 or might not but if it does fail, Microsoft will not provide any form of support.
> 
> Let me know if you have another question.

@teo-tsirpanis What do i do next?

**Reply by teo-tsirpanis** [reply]
I don't know, what were you trying to do with .NET 8 and Windows 7 in the first place?

**Reply by colejohnson66** [reply]
It means that it works. How you use it will be detailed on their repository, not .NET’s.

**Reply by Sin-Shadow-Fox** [reply]
> I don't know, what were you trying to do with .NET 8 and Windows 7 in the first place?

@teo-tsirpanis I was trying to download Firmament

**Reply by colejohnson66** [reply]
Again, how you do that is in DepotDownloader's documentation.

**Reply by Sin-Shadow-Fox** [reply]
@teo-tsirpanis

**Reply by teo-tsirpanis** [reply]
@Sin-Shadow-Fox I believe the original question about the necessity of the Visual C++ redistributables has been answered.

For any additional assistance with the use of DepotDownloader, you should direct your questions to [that repository](https://github.com/SteamRE/DepotDownloader). Thanks.
